dgl_ptm/dgl_ptm/agent/agent_update.py

```diff
@@ -15,7 +15,7 @@ def agent_update(model_graph, model_params, model_data=None, timestep=None, meth
     elif method == 'consumption':
         wealth_consumption(model_graph, model_params,method=model_params['consume_method'])
     elif method == 'income':
-        income_generation(model_graph,model_params,method=model_params['income_method'])
+        income_generation(model_graph, device, model_params,method=model_params['income_method'])
```

Comment:
can you please add an inline comment above this line and explain why there is an argument `device` for this method and not other methods?

Reply:
Added.